lib/mayaHydra/hydraExtensions/mayaHydraLibInterface.h

```diff
@@ -26,7 +26,7 @@
 
 PXR_NAMESPACE_OPEN_SCOPE
 
-using SceneIndicesVector                                        = std::vector<HdSceneIndexBasePtr>;//Be careful, these are not not Ref counted. Elements could become dangling
+using SceneIndicesVector = std::vector<HdSceneIndexBaseRefPtr>;
```

Comment:
See this comment and discussion : https://github.com/Autodesk/maya-hydra/pull/4#discussion_r1394377638. Doing this now because I added a utility observer class along with the unit test for this issue, and that class needs its observed SceneIndex to remain valid for its whole lifetime.